### ISSUE TITLE
useMemo instead of useCallback

### ISSUE THREAD

**Issue by JoeyDjr** [okp4/ui#395]
The purpose is to use the `useMemo` react hook instead of `useCallback` when it is better.

> the difference between useMemo and useCallback is that useMemo returns the memo-ized value returned by the passed function. It only recalculates the value when one of the dependencies changes. It’s very useful if you want to avoid expensive calculations on every render when the returned value isn’t changing.

[Differences between usememo and useCallback](https://blog.shahednasser.com/react-usememo-vs-usecallback-when-to-use/#:~:text=useMemo%20is%20very%20similar%20to,one%20of%20the%20dependencies%20changes.)

The React docs say that useCallback returns a memoized callback and that useMemo returns a memoized value.
In other words, useCallback gives you referential equality between renders for functions. And useMemo gives you referential equality between renders for values.

[useMemo vs useCallback](https://medium.com/@jan.hesters/usecallback-vs-usememo-c23ad1dc60)
